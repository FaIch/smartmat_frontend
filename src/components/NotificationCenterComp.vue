<template>
    <div id="app">
        <div class="notification-button-container">
            <button class="notification-button" @click="toggleNotifications">
                <img src="../assets/icons/notification.svg" alt="NotificationBell" />
            </button>
            <div v-if="showNotifications" class="notifications-popup">
                <NotificationView />
            </div>
        </div>
    </div>
</template>

<script setup lang="ts">
import { ref } from 'vue'
import NotificationView from '../views/NotificationView.vue'

const showNotifications = ref(false)

const toggleNotifications = () => {
  showNotifications.value = !showNotifications.value
  console.log(showNotifications.value)
}
</script>

<style scoped>
.notification-button {
  padding-left: 10px;
  background-color: #1A7028;
  border: none;
  padding-bottom: 10px;
}

.notification-button img {
    height: 30px;
    width: 30px;
}

.notifications-popup {
    position: absolute;
    top: 100%;
    /* Set the top to 100% to position the popup right below the notification button */
    left: 60%;
    transform: translateX(-57%);
    width: 600px;
    height: 700px;
    z-index: 1000;
    background-color: white;
    overflow: scroll;
    scrollbar-width: thin;
}

  .notifications-popup::-webkit-scrollbar-thumb {
    background-color: #1A7028;
    width: 6px;
  }

  @media (max-width: 760px) {
    .notifications-popup {
    position: absolute;
    top: 100%;
    /* Set the top to 100% to position the popup right below the notification button */
    left: 60%;
    transform: translateX(-57%);
    width: 80%;
    height: 700px;
    z-index: 1000;
    background-color: white;
    overflow: scroll;
    scrollbar-width: 10px;
  }
}
</style>
