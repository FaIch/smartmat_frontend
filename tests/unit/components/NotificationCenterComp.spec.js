import { mount } from '@vue/test-utils'
import { describe, expect, it, beforeEach } from 'vitest'
import NotificationCenterComp from '../../../src/components/NotificationCenterComp.vue'
import { createPinia } from 'pinia'

const pinia = createPinia()
const mockStore = {
  someGetter: 'mockValue'
}
let wrapper

beforeEach(() => {
  wrapper = mount(NotificationCenterComp, {
    global: {
      plugins: [pinia],
      mocks: {
        $pinia: pinia,
        $store: mockStore
      }
    }
  })
})

describe('NotificationCenterComp form', async () => {
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('hides notification view by default', async () => {
    expect(wrapper.vm.showNotifications).toBe(false)
  })
})
