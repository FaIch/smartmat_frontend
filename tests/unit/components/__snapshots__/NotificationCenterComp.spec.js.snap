// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavbarComp form > renders correctly 1`] = `
"<div data-v-9ecb1c94=\\"\\" id=\\"app\\">
  <div data-v-9ecb1c94=\\"\\" class=\\"notification-button-container\\"><button data-v-9ecb1c94=\\"\\" class=\\"notification-button\\"><img data-v-9ecb1c94=\\"\\" src=\\"/src/assets/icons/notification.svg\\" alt=\\"NotificationBell\\"></button>
    <!--v-if-->
  </div>
</div>"
`;
